Adds current EWS/BG frequency to mobile pages

DaeronAlagos opened the pull request:
[O2345] Adds additional information of the current associated frequency of EWS & Blood Glucose observations to tasks cards (EWS & BG Obs only), patient cards and escalation forms for the respective observation types.

Commit:
Fixes AttributeError when context does not exist

No associated task. Found this in a specific scenario.

Commit:
Adds current EWS/BG frequency to mobile pages

[O2345] Adds additional information of the current associated frequency of EWS & Blood Glucose observations to tasks cards (EWS & BG Obs only), patient cards and escalation forms for the respective observation types.

PeterAlabaster commented on nh_eobs/sql_statements.py:
```diff
@@ -459,10 +463,8 @@ def get_collect_activities_sql(self, activity_ids_sql):
-                else 0
-            end as frequency,
+            coalesce(ews0.frequency, 0) as frequency,
+            coalesce(bg0.frequency, 0) as bg_frequency,
```
why is coalesce used here? This suggests there's more than 1 row returned by ews0 which could be problematic

PeterAlabaster replied:
i'm an idiot, misread coalesce as concatenate

PeterAlabaster commented on nh_eobs_mobile/controllers/main.py:
```diff
@@ -575,6 +575,16 @@ def get_patients(self, *args, **kw):
 
+        for patient in patients:
+            if patient.get('frequency'):
+                patient['frequency_string'] = '{:2d} hour(s) {:02d} min(s)'.format(
```
```suggestion
                patient['frequency_string'] = self.minutes_to_hours_minutes_string(patient.get('frequency'))

    def minutes_to_hours_minutes_string(self, minutes):
        return '{:2d} hour(s) {:02d} min(s)'.format(*divmod(minutes, 60))
```

function name is up to you

PeterAlabaster commented on nh_eobs_mobile/controllers/main.py:
```diff
@@ -575,6 +575,16 @@ def get_patients(self, *args, **kw):
+                    *divmod(patient.get('frequency'), 60)
+                )
+            if patient.get('bg_frequency'):
+                patient['bg_frequency_string'] = '{:2d} hour(s) {:02d} min(s)'.format(
```
call to function here

PeterAlabaster commented on nh_eobs_mobile/controllers/main.py:
```diff
@@ -634,6 +651,23 @@ def process_escalations(self, creator_id, *args, **kwargs):
+    @staticmethod
+    def _get_ews_frequency(creator_activity):
+        if creator_activity.data_model == 'nh.clinical.patient.observation.ews':
+            return '{:2d} hour(s) {:02d} min(s)'.format(
```
call to function here

PeterAlabaster commented on nh_eobs_mobile/controllers/main.py:
```diff
@@ -634,6 +651,23 @@ def process_escalations(self, creator_id, *args, **kwargs):
+    def _get_bg_frequency(creator_activity):
+        if creator_activity.data_model == \
+                'nh.clinical.patient.observation.blood_glucose':
+            return '{:2d} hour(s) {:02d} min(s)'.format(
```
call to function here

PeterAlabaster commented on nh_eobs_mobile/controllers/main.py:
```diff
@@ -868,6 +902,16 @@ def get_tasks(self, *args, **kw):
 
+        for task in tasks:
+            if task.get('frequency'):
+                task['frequency_string'] = '{:2d} hour(s) {:02d} min(s)'.format(
```
call to function here

PeterAlabaster commented on nh_eobs_mobile/controllers/main.py:
```diff
@@ -868,6 +902,16 @@ def get_tasks(self, *args, **kw):
+                    *divmod(task.get('frequency'), 60)
+                )
+            if task.get('bg_frequency'):
+                task['bg_frequency_string'] = '{:2d} hour(s) {:02d} min(s)'.format(
```
call to function here

Commit:
Refactor to remove duplicate code

[O2345] Adds additional information of the current associated frequency of EWS & Blood Glucose observations to tasks cards (EWS & BG Obs only), patient cards and escalation forms for the respective observation types.
